target/riscv/op_helper_cheri.c

```diff
@@ -82,35 +94,38 @@ struct SCRInfo {
+    [CheriSCR_UTCC] = {.r = true, .w = true, .access = U_ASR_RW, .name = "UTCC"},
+    [CheriSCR_UTDC] = {.r = true, .w = true, .access = U_ASR_RW, .name = "UTDC"},
     [CheriSCR_UScratchC] = {.r = true,
                             .w = true,
-                            .access = U_ASR,
+                            .access = U_ASR_RW,
                             .name = "UScratchC"},
-    [CheriSCR_UEPCC] = {.r = true, .w = true, .access = U_ASR, .name = "UEPCC"},
+    [CheriSCR_UEPCC] = {.r = true, .w = true, .access = U_ASR_RW, .name = "UEPCC"},
+    [CheriSCR_UTIDC] = {.r = true, .w = true, .access = U_ASR_W, .name = "UTIDC"},
 
-    [CheriSCR_STCC] = {.r = true, .w = true, .access = S_ASR, .name = "STCC"},
-    [CheriSCR_STDC] = {.r = true, .w = true, .access = S_ASR, .name = "STDC"},
+    [CheriSCR_STCC] = {.r = true, .w = true, .access = S_ASR_RW, .name = "STCC"},
+    [CheriSCR_STDC] = {.r = true, .w = true, .access = S_ASR_RW, .name = "STDC"},
     [CheriSCR_SScratchC] = {.r = true,
                             .w = true,
-                            .access = S_ASR,
+                            .access = S_ASR_RW,
                             .name = "SScratchC"},
-    [CheriSCR_SEPCC] = {.r = true, .w = true, .access = S_ASR, .name = "SEPCC"},
+    [CheriSCR_SEPCC] = {.r = true, .w = true, .access = S_ASR_RW, .name = "SEPCC"},
+    [CheriSCR_STIDC] = {.r = true, .w = true, .access = S_ASR_W, .name = "STIDC"},
 
-    [CheriSCR_MTCC] = {.r = true, .w = true, .access = M_ASR, .name = "MTCC"},
-    [CheriSCR_MTDC] = {.r = true, .w = true, .access = M_ASR, .name = "MTDC"},
+    [CheriSCR_MTCC] = {.r = true, .w = true, .access = M_ASR_RW, .name = "MTCC"},
+    [CheriSCR_MTDC] = {.r = true, .w = true, .access = M_ASR_RW, .name = "MTDC"},
     [CheriSCR_MScratchC] = {.r = true,
                             .w = true,
-                            .access = M_ASR,
+                            .access = M_ASR_RW,
                             .name = "MScratchC"},
-    [CheriSCR_MEPCC] = {.r = true, .w = true, .access = M_ASR, .name = "MEPCC"},
+    [CheriSCR_MEPCC] = {.r = true, .w = true, .access = M_ASR_RW, .name = "MEPCC"},
+    [CheriSCR_MTIDC] = {.r = true, .w = true, .access = M_ASR_W, .name = "MTIDC"},
 
-    [CheriSCR_BSTCC] = {.r = true, .w = true, .access = H_ASR, .name= "BSTCC"},
-    [CheriSCR_BSTDC] = {.r = true, .w = true, .access = H_ASR, .name= "BSTCC"},
-    [CheriSCR_BSScratchC] = {.r = true, .w = true, .access = H_ASR,
-                             .name= "BSTCC"},
-    [CheriSCR_BSEPCC] = {.r = true, .w = true, .access = H_ASR, .name= "BSTCC"},
+    [CheriSCR_BSTCC] = {.r = true, .w = true, .access = H_ASR_RW, .name = "BSTCC"},
+    [CheriSCR_BSTDC] = {.r = true, .w = true, .access = H_ASR_RW, .name = "BSTCC"},
+    [CheriSCR_BSScratchC] = {.r = true, .w = true, .access = H_ASR_RW,
+                             .name = "BSTCC"},
+    [CheriSCR_BSEPCC] = {.r = true, .w = true, .access = H_ASR_RW, .name = "BSTCC"},
```

Comment:
Most of these lines are more than 80 characters long, which is causing CI to complain.